### PR TITLE
Allow to install outside of Joomla

### LINKED ISSUE #50: Allow to install outside of Joomla
Ensure that the composer install process can run outside of the Joomla context, this would allow installing the  Joomlatools Framework standalone.

### PR DESCRIPTION
**Pull request by yiendos**
This PR closes #50 

## Note to self how to run composer using local repository

```
composer config repositories.composer '{"type": "path", "url": "/Users/paulyiend/Sites/joomlatools-vagrant/Projects/joomlatools-vagrant/Projects/joomlatools-composer", "trunk-path": "feature/50-standalone"}'

composer require joomlatools/composer:"dev-feature/50-standalone"
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,6 +19,6 @@
         "class": "Joomlatools\\Composer\\Plugin"
     },
     "require": {
-        "composer-plugin-api": "^1.0"
+        "composer-plugin-api": "^2.0"
```
Comment (johanjanssens) [inline composer.json:22]: @yiendos Should we do the upgrade to the 2.0 API in a diff ticket?

File: src/Joomlatools/Joomla/Util.php [hunk 2]
```diff
@@ -24,6 +24,27 @@ public static function getPlatformName()
         return static::isJoomlatoolsPlatform() ? 'Joomlatools Platform' : 'Joomla';
     }
 
+    /**
+     * Validate if the current working directory is a Standalone app
+     *
+     * @return bool
+     */
+    public static function isStandalone()
```
Comment (johanjanssens) [inline src/Joomlatools/Joomla/Util.php:32]: @yiendos I'm not so sure about this check. I would do this different, the script is running standalone if it's not running in Joomla or in platform.

So maybe the solution is to not add a isStandalone() method but instead change the code, and do not do further installation processing by checking

`if (!Util::isJoomla() && !Util::isJoomlatoolsPlatform())`  

Thoughts?

**Review by johanjanssens — CHANGES_REQUESTED** [review]

File: src/Joomlatools/Joomla/Util.php (hunk 2 above)
Comment (yiendos) [reply src/Joomlatools/Joomla/Util.php]: No makes sense and makes the code similar to other checks in the joomlatools-composer plugin

**Comment by yiendos** [comment]
@johanjanssens requested changes have been made, joomlatools-composer, joomlatools-framework, joomlatools-pages are now installable via composer. In so far as they get installed into the vendor folder. Our discussions on Slack you mentioned that you wanted to update the symlinks in the repository to point to the vendor folder itself. 

Or like foliokit you could point to the composer autoloader. 

Now it seems to me: 

1. If you wanted composer to create these symlinks automatically as part of a post-autoload hook:
https://github.com/joomlatools/joomlatools-composer/pull/51/files#diff-b9a6a3e3498e7dee123756c15e44b65bdff1ca9b8cb592679e35cbd64f28b049L71

Like we do for traditional J! routes, joomlatools-composer would be an ideal candidate. 

2. If you wanted to create these symlinks manually,  the integration into joomlatools-composer seems un-warranted. A simple `composer require joomlatools/framework` would suffice. 

Let me know which way you'd like to proceed with this.

**Review by johanjanssens — APPROVED** [review]

**Comment by johanjanssens** [comment]
@yiendos as part of this change please remove the upgrade to Composer 2. Will handle that in a separate ticket.

**Comment by yiendos** [comment]
@johanjanssens requested changes have been made, we've rolled back to composer-api ^1 and I've tested the standalone solution with www.joomlatools.com and tested the existing integration to a traditional J! platform. Let me know your thoughts

**Pull request merged**